### PR TITLE
Add LinkedIn embed

### PR DESCRIPTION
**Pull request by PrikshitSingh24**
added linkedIn embed which looks something like this:
Screenshot from 2023-10-04 23-51-46

### CODE REVIEW

File: shared/editor/embeds/index.tsx [hunk 1]
```diff
@@ -298,6 +299,13 @@ const embeds: EmbedDescriptor[] = [
     icon: <Img src="/images/jsfiddle.png" alt="JSFiddle" />,
     component: JSFiddle,
   }),
+  new EmbedDescriptor({
+    title: "Linkedin",
+    keywords: "post",
+    defaultHidden: true,
+    icon: <Img src="/images/linkedin.png" alt="Linkedin" />,
```
Comment (tommoor) [inline shared/editor/embeds/index.tsx:306]: Missing the associated 120x120 image here

**Comment by PrikshitSingh24** [comment]
I have accepted the CLA but it is still showing agreement not signed, do i need to do something else?

**Comment by tommoor** [comment]
Looks like it's stuck, it happens – no worries as it's a non-blocking check

**Comment by tommoor** [comment]
I don't love that while loop, this is immediately the most complex embed in the system which makes me think that logic can be avoided. In `LinkedIn.ENABLED` it's possible to specify multiple regex which might simplify and allow writing tests.

Do you have some examples of urls that should match here? I'm not familiar with LinkedIn

**Comment by PrikshitSingh24** [comment]
Yes, some examples of urls : https://www.linkedin.com/posts/github_asakawa-activity-7117896142612267009-Cg_x?utm_source=share&utm_medium=member_desktop , https://www.linkedin.com/posts/professional-development-harvard-division-of-continuing-education_online-programs-from-harvard-activity-7031651466717102081-2hc4?utm_source=share&utm_medium=member_desktop

**Comment by zeekhuge** [comment]
@PrikshitSingh-AL Whats the difference between those 2 urls, ie, between the following URLs
1. https://www.linkedin.com/posts/github_asakawa-activity-7117896142612267009-Cg_x
2. https://www.linkedin.com/posts/jaredspa_generativeai-aiskills-futureofwork-ugcPost-7113555729428480000-dVrv

They look exactly the same (in terms of URL structure).

File: shared/editor/embeds/index.tsx (hunk 1 above)
Comment (tommoor) [inline shared/editor/embeds/index.tsx:306]:
```suggestion
    icon: <Img src="/images/linkedin.png" alt="LinkedIn" />,
```

File: shared/editor/embeds/index.tsx [hunk 2]
```diff
@@ -298,6 +299,13 @@ const embeds: EmbedDescriptor[] = [
     icon: <Img src="/images/jsfiddle.png" alt="JSFiddle" />,
     component: JSFiddle,
   }),
+  new EmbedDescriptor({
+    title: "Linkedin",
```
Comment (tommoor) [inline shared/editor/embeds/index.tsx:303]:
```suggestion
    title: "LinkedIn",
```

File: shared/editor/embeds/Linkedin.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  const objectId = matches[2];
+  const postType = matches[1];
+  if (matches[3] === "embed") {
+    return <Frame {...props} src={`${props.attrs.href}`} title="Linkedin" />;
+  }
+  return (
+    <Frame
+      {...props}
+      src={`https://www.linkedin.com/embed/feed/update/urn:li:${postType}:${objectId}`}
+      title="Linkedin"
```
Comment (tommoor) [inline shared/editor/embeds/Linkedin.tsx:16]:
```suggestion
      title="LinkedIn"
```

File: shared/editor/embeds/Linkedin.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+import Frame from "../components/Frame";
+import { EmbedProps as Props } from ".";
+
+function Linkedin(props: Props) {
+  const { matches } = props.attrs;
+  const objectId = matches[2];
+  const postType = matches[1];
+  if (matches[3] === "embed") {
+    return <Frame {...props} src={`${props.attrs.href}`} title="Linkedin" />;
```
Comment (tommoor) [inline shared/editor/embeds/Linkedin.tsx:10]:
```suggestion
    return <Frame {...props} src={props.attrs.href} title="LinkedIn" />;
```

**Review by tommoor — COMMENTED** [review]
Looks much better now, just need to fix the name – it's styled with a capitalized "I"

**Review by tommoor — APPROVED** [review]

**Pull request merged**